lib/matplotlib/axes/_axes.py

```diff
@@ -5585,6 +5585,11 @@ def imshow(self, X, cmap=None, norm=None, *, aspect=None,
               that the data fit in the Axes. In general, this will result in
               non-square pixels.
 
+            This parameter is ignored if the image uses a transform that does
```

Comment:
Is this really what we want?  If you do this then `aspect=2` no longer will work.   This seems like such an edge case, I'd rather just force users to pass `aspect='auto'`, eg keep the current state.